### DISCUSSION TITLE
Strange behavior of subdomain

### DISCUSSION THREAD

**Discussion by empx92** [kimai/kimai#3944] [On-Premise: Installation & Updates]
Im already using Kimai 1.30 on my Webserver and now installed 2.0.11 on the same Webserver and having some troubles here.

For example:
1.30 is in my directory /html/zeiterfassung so my subdomain points to /zeiterfassung with an own created index.php which is forwarding to /zeiterfassung/public which works fine for the 1.30 Installation (i used this solution because there was a Problem when pointing directly to /zeiterfassung/public/)

So for the Version 2.0.11 which is installed in the directory /html/time the subdomain should point to /time/public/ which throws an error:

`[23-Mar-2023 15:38:42 UTC] PHP Warning:  require(): open_basedir restriction in effect. File(/data/web/e115***/html/time/vendor/composer/platform_check.php) is not within the allowed path(s): (/data/web/e115***/html/time/public:/data/web/e115***/phpsession:/data/web/e115***/phptmp:/tmp:/usr/bin) in /data/web/e115***/html/time/vendor/composer/autoload_real.php on line 25
[23-Mar-2023 15:38:42 UTC] PHP Warning:  require(/data/web/e115***/html/time/vendor/composer/platform_check.php): Failed to open stream: Operation not permitted in /data/web/e115***/html/time/vendor/composer/autoload_real.php on line 25
[23-Mar-2023 15:38:42 UTC] PHP Fatal error:  Uncaught Error: Failed opening required '/data/web/e115***/html/time/vendor/composer/platform_check.php' (include_path='.') in /data/web/e115***/html/time/vendor/composer/autoload_real.php:25
Stack trace:
#0 /data/web/e115***/html/time/vendor/autoload.php(25): ComposerAutoloaderInit93761b1f3afb658b8a8a3ecc8eff1c43::getLoader()
#1 /data/web/e115***/html/time/vendor/autoload_runtime.php(5): require_once('/data/web/e1151...')
#2 /data/web/e115***/html/time/public/index.php(5): require_once('/data/web/e1151...')
#3 {main}
  thrown in /data/web/e115***/html/time/vendor/composer/autoload_real.php on line 25`
  
  It only works if the subdomain points to /time but then the sources aren't loaded correct because for example the browser tries to load "https://time2.subdomain.com/bundles/tabler/fonts/fa-regular-400.woff2?7a333762" would be correctly located in "https://time2.subdomain.com/**public/**bundles/tabler/fonts/fa-regular-400.woff2?7a333762"
  
  Does anyone know a solution for that problem?
  

best regards

**Comment by kevinpapst** [comment]
Kimai never officially supported being hosted in a sub-directory. If it worked for 1.x, you were lucky 👍 

The log about `open_basedir` is an environment problem. 
If you point your document root to /public and then your hoster (or whatever you use to configure your server) sets the open_basedir to the same directory, it is simply incorrect.

Deactivate open_basedir or if you are on a shared hosting (which has always problems) at least configure it to kimai/ 
Ask your hoster how to do that.

**Reply by empx92** [reply]
thanks for your answer.
yes i read about that it is not officially possible to install it on a subdirectory but i simply can not order a new hosting ;)
i simply could run it on our server but i'm currently out of fix ip's so i tried to run the V2 same way than V1.3 ;)

Could you tell me where the "dir" is set so i could simply add the /public/ to the string or is it really a limitation which could not be edited? (i know "update = edit again")

best regards

**Reply by kevinpapst** [reply]
open_basedir is a PHP setting.
If you are on a shared host: good luck.
If you are on your own server, edit php.ini (Google/Bing will tell you how to do that)

**Reply by empx92** [reply]
thank you very much you were right that was simply the open_basedir conflict. which can be turned off really easy @easyname per subdomain or per PHP Setting in CP.

Thanks
